lib/utils/metrics.js

```diff
@@ -0,0 +1,146 @@
+'use strict';
+
+const assert = require('assert');
+const uuid = require('uuid');
+
+/** @typedef { import('bmutex').Lock } Lock */
+
+const metrics = exports;
+
+/**
+ * @typedef {Object} Honey
+ */
+
+class Event {
+  /**
+   *
+   * @param {string} name the name for the event
+   * @param {Event | Honey} parent a parent Event or a libhoney instance
+   */
+  constructor(name, parent) {
+    assert(name, 'Event name is required');
+    assert(parent, 'Parent event or libhoney instance is required.');
+    if (parent instanceof Event) {
+      this.honey = parent.honey;
+      this.parent = parent;
+      this.trace_id = parent.trace_id;
+    } else {
+      this.honey = parent;
+      this.parent = null;
+      this.trace_id = uuid.v4();
+    }
+
+    this.event = this.honey.newEvent();
```

Comment:
I think that we can easily make `this.honey` nullable if we provide a default stub when `this.honey` is not provided. It would be useful whenever we run hsd independent of our other infra that provides honey. Here is a minimal viable stub as far as I can tell:

```javascript
const stubbedEvent = { send: () => {}, addField: () => {} };
const honeyStub = { newEvent: () => stubbedEvent };
```

Then when we declare this class the constructor can just set `parent = honeyStub` and everything would work